src/ParquetMsg.chpl

```diff
@@ -454,10 +457,32 @@ module ParquetMsg {
           valPtr = c_ptrTo(locArr);
         }
         if mode == TRUNCATE || !filesExist {
-          if c_writeColumnToParquet(myFilename.localize().c_str(), valPtr, 0,
-                                    dsetname.localize().c_str(), locDom.size, rowGroupSize,
-                                    dtypeRep, compression, c_ptrTo(pqErr.errMsg)) == ARROWERROR {
-            pqErr.parquetError(getLineNumber(), getRoutineName(), getModuleName());
+          if !doParallel {
+            if c_writeColumnToParquet(myFilename.localize().c_str(), valPtr, 0,
+                                      dsetname.localize().c_str(), locDom.size, rowGroupSize,
+                                      dtypeRep, compression, c_ptrTo(pqErr.errMsg)) == ARROWERROR {
+              pqErr.parquetError(getLineNumber(), getRoutineName(), getModuleName());
+            }
+          } else {
+            var fileSizes: [0..#loc.maxTaskPar] int = locDom.size/loc.maxTaskPar;
+            // First file has the extra elements if it isn't evenly divisible by maxTaskPar
+            fileSizes[0] += locDom.size - ((locDom.size/loc.maxTaskPar)*loc.maxTaskPar);
```

Comment:
i had to convince myself that the integer cast of the float divide would always round down. This seems to check out and the adjustment gave what i expected on a small example!
```chapel
var fileSizes: [0..#6] int = 10/6;  // 1.6666 verify this rounds down
writeln(fileSizes);
var leftOver = 10 - ((10/6)*6); 
writeln(leftOver);
```
```
1 1 1 1 1 1
4
```

I will say in my small example this resulted in a pretty unbalanced distribution, but I *think* that in a real case that `locDom.size` would be large enough relative to `loc.maxTaskPar` that it would be pretty uniform... I'm just now realizing that's probably part of the motivation for having a `parallelWriteThreshold` lol

Reply:
Since `leftOver` the remaineder of `locDom.size/loc.maxTaskPar`, it should be guaranteed to be less than `fileSizes.size`. So if we wanted to distribute the remained more evenly we could do something like
```chapel
var fileSizes: [0..#6] int = 10/6;
writeln(fileSizes);
var leftOver = 10 - ((10/6)*6); 
writeln(leftOver);
fileSizes[0..#leftOver] += 1;
writeln(fileSizes);
```
```
1 1 1 1 1 1
4
2 2 2 2 1 1
```
But this is probably overengineering something that isn't actually a problem